raiden/transfer/utils.py

```diff
@@ -39,11 +39,13 @@ def get_state_change_with_balance_proof_by_balance_hash(
     has the blinded balance proof data.
     """
     return storage.get_latest_state_change_by_data_field({
-        'balance_proof.chain_id': canonical_identifier.chain_identifier,
-        'balance_proof.token_network_identifier': to_checksum_address(
+        'balance_proof.canonical_identifier.chain_identifier':
+        str(canonical_identifier.chain_identifier),
```

Comment:
Our `to_dict` serialization string-ifies the  int value (a good practice for json serializations, since implementations can differ in their number resolutions, afaiu).